packages/ui-test-queries/src/utils/bindElementToMethods.ts

```diff
@@ -21,16 +21,21 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-export function bindElementToMethods(element, methods) {
-  return Object.entries(methods).reduce((bound, [key, fn]) => {
-    if (typeof fn === 'function') {
-      bound[key] = fn.bind(null, element) // eslint-disable-line no-param-reassign
-      return bound
-    } else {
-      throw new Error(
-        `[ui-test-queries] cannot bind to a non-function of type ${typeof fn}`,
-        key
-      )
-    }
-  }, {})
+
+export type GenericFunction = (...args: any) => unknown | any
+
+export function bindElementToMethods<T, K>(element: T, methods: K) {
+  return Object.entries(methods).reduce(
+    (bound: Record<string, K>, [key, fn]: [string, K]) => {
+      if (typeof fn === 'function') {
+        bound[key] = fn.bind(null, element) // eslint-disable-line no-param-reassign
```

Comment:
This is why I could not type the added methods properly. This code adds the first parameter to the function automatically, I dont think we can remove it.